chess/game.go

```diff
@@ -0,0 +1,52 @@
+package chess
+
+// A Game contains the history of a chess game.
+type Game struct {
+	Positions []Position
+	Moves     []Move
+
+	// The number of half moves since the last capture or pawn advance.
+	HalfMoveClock int
+
+	// The full move number for the next move.
+	FullMoveNumber int
+}
+
+// NewGame returns a new game.
+func NewGame() *Game {
+	return &Game{
+		Positions:      []Position{NewPosition()},
+		Moves:          []Move{},
+		FullMoveNumber: 1,
+	}
+}
+
+// FirstPosition returns the first position of the game.
+func (g *Game) FirstPosition() Position {
+	return g.Positions[0]
+}
+
+// CurrentPosition returns the current game position.
+func (g *Game) CurrentPosition() Position {
+	return g.Positions[len(g.Positions)-1]
+}
+
+// Move applies a move to the game. The move must be legal.
+func (g *Game) Move(m Move) {
+	p := g.CurrentPosition()
+
+	if p.ActiveColor == Black {
+		g.FullMoveNumber++
+	}
+
+	reset := p.Move(m)
+
+	if reset {
+		g.HalfMoveClock = 0
+	} else {
+		g.HalfMoveClock++
+	}
```

Comment:
nit: can combine. probably still want a named variable for clarity though

```suggestion
	if reset := p.Move(m); reset {
		g.HalfMoveClock = 0
	} else {
		g.HalfMoveClock++
	}
```